// @prisma/client@5.22.0/.prisma/client/wasm.js

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 5.22.0
 * Query Engine version: 605197351a3c8bdd595af2d2a9bc3025bca48ea2
 */
Prisma.prismaVersion = {
  client: "5.22.0",
  engine: "605197351a3c8bdd595af2d2a9bc3025bca48ea2"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.NotFoundError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`NotFoundError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  email: 'email',
  password: 'password',
  name: 'name',
  isAdmin: 'isAdmin',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ProjectScalarFieldEnum = {
  id: 'id',
  name: 'name',
  apiKey: 'apiKey',
  userId: 'userId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.DeviceScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  deviceId: 'deviceId',
  platform: 'platform',
  osVersion: 'osVersion',
  appVersion: 'appVersion',
  model: 'model',
  manufacturer: 'manufacturer',
  metadata: 'metadata',
  lastSeenAt: 'lastSeenAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  deviceCode: 'deviceCode',
  userId: 'userId',
  userEmail: 'userEmail',
  userName: 'userName',
  debugModeEnabled: 'debugModeEnabled',
  debugModeEnabledAt: 'debugModeEnabledAt',
  debugModeExpiresAt: 'debugModeExpiresAt',
  debugModeEnabledBy: 'debugModeEnabledBy',
  status: 'status',
  deletedAt: 'deletedAt'
};

exports.Prisma.LogScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  deviceId: 'deviceId',
  sessionId: 'sessionId',
  level: 'level',
  message: 'message',
  tag: 'tag',
  data: 'data',
  fileName: 'fileName',
  lineNumber: 'lineNumber',
  functionName: 'functionName',
  className: 'className',
  screenName: 'screenName',
  threadName: 'threadName',
  timestamp: 'timestamp',
  createdAt: 'createdAt'
};

exports.Prisma.CrashScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  deviceId: 'deviceId',
  message: 'message',
  stackTrace: 'stackTrace',
  metadata: 'metadata',
  timestamp: 'timestamp',
  createdAt: 'createdAt'
};

exports.Prisma.ApiTraceScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  deviceId: 'deviceId',
  sessionId: 'sessionId',
  url: 'url',
  method: 'method',
  statusCode: 'statusCode',
  requestHeaders: 'requestHeaders',
  requestBody: 'requestBody',
  responseHeaders: 'responseHeaders',
  responseBody: 'responseBody',
  duration: 'duration',
  error: 'error',
  screenName: 'screenName',
  networkType: 'networkType',
  country: 'country',
  carrier: 'carrier',
  ipAddress: 'ipAddress',
  userAgent: 'userAgent',
  cost: 'cost',
  timestamp: 'timestamp',
  createdAt: 'createdAt'
};

exports.Prisma.ApiConfigScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  endpoint: 'endpoint',
  method: 'method',
  name: 'name',
  description: 'description',
  costPerRequest: 'costPerRequest',
  isEnabled: 'isEnabled',
  enableLogs: 'enableLogs',
  captureRequestBody: 'captureRequestBody',
  captureResponseBody: 'captureResponseBody',
  metadata: 'metadata',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SessionScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  deviceId: 'deviceId',
  sessionToken: 'sessionToken',
  startedAt: 'startedAt',
  endedAt: 'endedAt',
  isActive: 'isActive',
  appVersion: 'appVersion',
  osVersion: 'osVersion',
  locale: 'locale',
  timezone: 'timezone',
  networkType: 'networkType',
  screenFlow: 'screenFlow',
  entryScreen: 'entryScreen',
  exitScreen: 'exitScreen',
  duration: 'duration',
  screenCount: 'screenCount',
  eventCount: 'eventCount',
  errorCount: 'errorCount',
  metadata: 'metadata',
  userProperties: 'userProperties',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.NotificationSettingsScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  emailEnabled: 'emailEnabled',
  emailAddresses: 'emailAddresses',
  pushEnabled: 'pushEnabled',
  pushToken: 'pushToken',
  smsEnabled: 'smsEnabled',
  smsNumbers: 'smsNumbers',
  webhookEnabled: 'webhookEnabled',
  webhookUrl: 'webhookUrl',
  webhookSecret: 'webhookSecret',
  webhookHeaders: 'webhookHeaders',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ApiAlertScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  title: 'title',
  description: 'description',
  endpoint: 'endpoint',
  method: 'method',
  isEnabled: 'isEnabled',
  monitorStandardErrors: 'monitorStandardErrors',
  standardErrorCodes: 'standardErrorCodes',
  customStatusCodes: 'customStatusCodes',
  bodyErrorField: 'bodyErrorField',
  bodyErrorValues: 'bodyErrorValues',
  headerErrorField: 'headerErrorField',
  headerErrorValues: 'headerErrorValues',
  notifyEmail: 'notifyEmail',
  notifyPush: 'notifyPush',
  notifySms: 'notifySms',
  notifyWebhook: 'notifyWebhook',
  cooldownMinutes: 'cooldownMinutes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MonitoredErrorScalarFieldEnum = {
  id: 'id',
  alertId: 'alertId',
  projectId: 'projectId',
  errorType: 'errorType',
  errorCode: 'errorCode',
  endpoint: 'endpoint',
  method: 'method',
  statusCode: 'statusCode',
  requestBody: 'requestBody',
  responseBody: 'responseBody',
  firstOccurrence: 'firstOccurrence',
  lastOccurrence: 'lastOccurrence',
  occurrenceCount: 'occurrenceCount',
  affectedDevices: 'affectedDevices',
  affectedSessions: 'affectedSessions',
  lastTraceId: 'lastTraceId',
  lastNotifiedAt: 'lastNotifiedAt',
  notificationCount: 'notificationCount',
  isResolved: 'isResolved',
  resolvedAt: 'resolvedAt',
  resolvedBy: 'resolvedBy',
  notes: 'notes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.BusinessConfigScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  key: 'key',
  label: 'label',
  description: 'description',
  valueType: 'valueType',
  stringValue: 'stringValue',
  integerValue: 'integerValue',
  booleanValue: 'booleanValue',
  decimalValue: 'decimalValue',
  jsonValue: 'jsonValue',
  imageUrl: 'imageUrl',
  category: 'category',
  isEnabled: 'isEnabled',
  version: 'version',
  metadata: 'metadata',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.UploadedFileScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  filename: 'filename',
  storedName: 'storedName',
  mimeType: 'mimeType',
  size: 'size',
  url: 'url',
  uploadedBy: 'uploadedBy',
  createdAt: 'createdAt'
};

exports.Prisma.LanguageScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  code: 'code',
  name: 'name',
  nativeName: 'nativeName',
  isDefault: 'isDefault',
  isEnabled: 'isEnabled',
  isRTL: 'isRTL',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.LocalizationKeyScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  key: 'key',
  description: 'description',
  category: 'category',
  platform: 'platform',
  maxLength: 'maxLength',
  screenshot: 'screenshot',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TranslationScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  keyId: 'keyId',
  languageId: 'languageId',
  value: 'value',
  isReviewed: 'isReviewed',
  reviewedBy: 'reviewedBy',
  reviewedAt: 'reviewedAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.FeatureFlagsScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  sdkEnabled: 'sdkEnabled',
  apiTracking: 'apiTracking',
  screenTracking: 'screenTracking',
  crashReporting: 'crashReporting',
  logging: 'logging',
  deviceTracking: 'deviceTracking',
  sessionTracking: 'sessionTracking',
  businessConfig: 'businessConfig',
  localization: 'localization',
  offlineSupport: 'offlineSupport',
  batchEvents: 'batchEvents',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SdkSettingsScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  trackingMode: 'trackingMode',
  captureRequestBodies: 'captureRequestBodies',
  captureResponseBodies: 'captureResponseBodies',
  capturePrintStatements: 'capturePrintStatements',
  sanitizeSensitiveData: 'sanitizeSensitiveData',
  sensitiveFieldPatterns: 'sensitiveFieldPatterns',
  maxLogQueueSize: 'maxLogQueueSize',
  maxTraceQueueSize: 'maxTraceQueueSize',
  flushIntervalSeconds: 'flushIntervalSeconds',
  enableBatching: 'enableBatching',
  minLogLevel: 'minLogLevel',
  verboseErrors: 'verboseErrors',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ConfigCategoryScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  name: 'name',
  label: 'label',
  description: 'description',
  icon: 'icon',
  order: 'order',
  isEnabled: 'isEnabled',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.PlanScalarFieldEnum = {
  id: 'id',
  name: 'name',
  displayName: 'displayName',
  description: 'description',
  price: 'price',
  currency: 'currency',
  interval: 'interval',
  isActive: 'isActive',
  isPublic: 'isPublic',
  maxProjects: 'maxProjects',
  maxDevices: 'maxDevices',
  maxMockEndpoints: 'maxMockEndpoints',
  maxApiEndpoints: 'maxApiEndpoints',
  maxApiRequests: 'maxApiRequests',
  maxLogs: 'maxLogs',
  maxSessions: 'maxSessions',
  maxCrashes: 'maxCrashes',
  maxBusinessConfigKeys: 'maxBusinessConfigKeys',
  maxLocalizationLanguages: 'maxLocalizationLanguages',
  maxLocalizationKeys: 'maxLocalizationKeys',
  retentionDays: 'retentionDays',
  allowApiTracking: 'allowApiTracking',
  allowScreenTracking: 'allowScreenTracking',
  allowCrashReporting: 'allowCrashReporting',
  allowLogging: 'allowLogging',
  allowBusinessConfig: 'allowBusinessConfig',
  allowLocalization: 'allowLocalization',
  allowCustomDomains: 'allowCustomDomains',
  allowWebhooks: 'allowWebhooks',
  allowTeamMembers: 'allowTeamMembers',
  allowPrioritySupport: 'allowPrioritySupport',
  features: 'features',
  enforcementConfig: 'enforcementConfig',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SubscriptionScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  planId: 'planId',
  status: 'status',
  enabled: 'enabled',
  trialStartDate: 'trialStartDate',
  trialEndDate: 'trialEndDate',
  currentPeriodStart: 'currentPeriodStart',
  currentPeriodEnd: 'currentPeriodEnd',
  cancelledAt: 'cancelledAt',
  cancelledReason: 'cancelledReason',
  disabledBy: 'disabledBy',
  disabledAt: 'disabledAt',
  enabledBy: 'enabledBy',
  enabledAt: 'enabledAt',
  quotaMaxProjects: 'quotaMaxProjects',
  quotaMaxDevices: 'quotaMaxDevices',
  quotaMaxMockEndpoints: 'quotaMaxMockEndpoints',
  quotaMaxApiEndpoints: 'quotaMaxApiEndpoints',
  quotaMaxApiRequests: 'quotaMaxApiRequests',
  quotaMaxLogs: 'quotaMaxLogs',
  quotaMaxSessions: 'quotaMaxSessions',
  quotaMaxCrashes: 'quotaMaxCrashes',
  quotaMaxBusinessConfigKeys: 'quotaMaxBusinessConfigKeys',
  quotaMaxLocalizationLanguages: 'quotaMaxLocalizationLanguages',
  quotaMaxLocalizationKeys: 'quotaMaxLocalizationKeys',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.EnforcementStateScalarFieldEnum = {
  id: 'id',
  subscriptionId: 'subscriptionId',
  state: 'state',
  warnEnteredAt: 'warnEnteredAt',
  graceEnteredAt: 'graceEnteredAt',
  graceEndsAt: 'graceEndsAt',
  degradedEnteredAt: 'degradedEnteredAt',
  effectivePolicy: 'effectivePolicy',
  triggeredMetrics: 'triggeredMetrics',
  lastEvaluatedAt: 'lastEvaluatedAt',
  nextEvaluationAt: 'nextEvaluationAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.InvoiceScalarFieldEnum = {
  id: 'id',
  subscriptionId: 'subscriptionId',
  invoiceNumber: 'invoiceNumber',
  status: 'status',
  amount: 'amount',
  currency: 'currency',
  periodStart: 'periodStart',
  periodEnd: 'periodEnd',
  dueDate: 'dueDate',
  paidAt: 'paidAt',
  pdfUrl: 'pdfUrl',
  lineItems: 'lineItems',
  notes: 'notes',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.BuildScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  version: 'version',
  name: 'name',
  description: 'description',
  mode: 'mode',
  isActive: 'isActive',
  createdBy: 'createdBy',
  createdAt: 'createdAt',
  businessConfigSnapshot: 'businessConfigSnapshot',
  localizationSnapshot: 'localizationSnapshot',
  configCount: 'configCount',
  translationCount: 'translationCount'
};

exports.Prisma.BuildFeatureScalarFieldEnum = {
  id: 'id',
  buildId: 'buildId',
  featureType: 'featureType',
  snapshotData: 'snapshotData',
  itemCount: 'itemCount',
  createdAt: 'createdAt'
};

exports.Prisma.BuildModeScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  previewBuildId: 'previewBuildId',
  productionBuildId: 'productionBuildId',
  updatedAt: 'updatedAt'
};

exports.Prisma.BuildChangeLogScalarFieldEnum = {
  id: 'id',
  buildId: 'buildId',
  featureType: 'featureType',
  changeType: 'changeType',
  itemKey: 'itemKey',
  itemLabel: 'itemLabel',
  oldValue: 'oldValue',
  newValue: 'newValue',
  changedBy: 'changedBy',
  changedAt: 'changedAt'
};

exports.Prisma.MockEnvironmentScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  name: 'name',
  description: 'description',
  basePath: 'basePath',
  mode: 'mode',
  whitelist: 'whitelist',
  blacklist: 'blacklist',
  isEnabled: 'isEnabled',
  isDefault: 'isDefault',
  createdBy: 'createdBy',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MockEndpointScalarFieldEnum = {
  id: 'id',
  environmentId: 'environmentId',
  path: 'path',
  method: 'method',
  description: 'description',
  isEnabled: 'isEnabled',
  order: 'order',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MockResponseScalarFieldEnum = {
  id: 'id',
  endpointId: 'endpointId',
  statusCode: 'statusCode',
  name: 'name',
  description: 'description',
  responseBody: 'responseBody',
  responseHeaders: 'responseHeaders',
  delay: 'delay',
  isDefault: 'isDefault',
  isEnabled: 'isEnabled',
  order: 'order',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MockConditionScalarFieldEnum = {
  id: 'id',
  responseId: 'responseId',
  endpointId: 'endpointId',
  type: 'type',
  key: 'key',
  operator: 'operator',
  value: 'value',
  isCaseSensitive: 'isCaseSensitive',
  order: 'order',
  createdAt: 'createdAt'
};

exports.Prisma.ConfigAlertScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  configId: 'configId',
  name: 'name',
  description: 'description',
  condition: 'condition',
  threshold: 'threshold',
  operator: 'operator',
  timeWindow: 'timeWindow',
  minOccurrences: 'minOccurrences',
  enabled: 'enabled',
  webhookUrl: 'webhookUrl',
  emailRecipients: 'emailRecipients',
  lastTriggered: 'lastTriggered',
  triggerCount: 'triggerCount',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ConfigAlertEventScalarFieldEnum = {
  id: 'id',
  alertId: 'alertId',
  projectId: 'projectId',
  configId: 'configId',
  severity: 'severity',
  message: 'message',
  metadata: 'metadata',
  acknowledged: 'acknowledged',
  acknowledgedBy: 'acknowledgedBy',
  acknowledgedAt: 'acknowledgedAt',
  createdAt: 'createdAt'
};

exports.Prisma.ConfigApprovalScalarFieldEnum = {
  id: 'id',
  projectId: 'projectId',
  configId: 'configId',
  changeType: 'changeType',
  changeData: 'changeData',
  status: 'status',
  requiredApprovals: 'requiredApprovals',
  currentApprovals: 'currentApprovals',
  approvers: 'approvers',
  approvals: 'approvals',
  requestedBy: 'requestedBy',
  requestedAt: 'requestedAt',
  decidedBy: 'decidedBy',
  decidedAt: 'decidedAt',
  decisionComment: 'decisionComment',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SystemConfigurationScalarFieldEnum = {
  id: 'id',
  category: 'category',
  key: 'key',
  value: 'value',
  encrypted: 'encrypted',
  description: 'description',
  isActive: 'isActive',
  metadata: 'metadata',
  updatedBy: 'updatedBy',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.JsonNullValueInput = {
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};


exports.Prisma.ModelName = {
  User: 'User',
  Project: 'Project',
  Device: 'Device',
  Log: 'Log',
  Crash: 'Crash',
  ApiTrace: 'ApiTrace',
  ApiConfig: 'ApiConfig',
  Session: 'Session',
  NotificationSettings: 'NotificationSettings',
  ApiAlert: 'ApiAlert',
  MonitoredError: 'MonitoredError',
  BusinessConfig: 'BusinessConfig',
  UploadedFile: 'UploadedFile',
  Language: 'Language',
  LocalizationKey: 'LocalizationKey',
  Translation: 'Translation',
  FeatureFlags: 'FeatureFlags',
  SdkSettings: 'SdkSettings',
  ConfigCategory: 'ConfigCategory',
  Plan: 'Plan',
  Subscription: 'Subscription',
  EnforcementState: 'EnforcementState',
  Invoice: 'Invoice',
  Build: 'Build',
  BuildFeature: 'BuildFeature',
  BuildMode: 'BuildMode',
  BuildChangeLog: 'BuildChangeLog',
  MockEnvironment: 'MockEnvironment',
  MockEndpoint: 'MockEndpoint',
  MockResponse: 'MockResponse',
  MockCondition: 'MockCondition',
  ConfigAlert: 'ConfigAlert',
  ConfigAlertEvent: 'ConfigAlertEvent',
  ConfigApproval: 'ConfigApproval',
  SystemConfiguration: 'SystemConfiguration'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }
        
        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
